models/vgg/vgg.hpp

```diff
@@ -0,0 +1,143 @@
+/**
+ * @file vgg.hpp
+ * @author Shubham Agrawal
+ * 
+ * Definition of VGG models.
+ * 
+ * For more information, kindly refer to the following paper.
+ * 
+ * Paper for VGG.
+ *
+ * @code
+ * @article{Simonyan2014,
+ *  author = {Karen Simonyan, Andrew Zisserman},
+ *  title = {Very Deep Convolutional Networks for Large-Scale Image Recognition},
+ *  year = {2014},
+ *  url = {https://arxiv.org/pdf/1409.1556.pdf}
+ * }
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MODELS_MODELS_VGG_VGG_HPP
+#define MODELS_MODELS_VGG_VGG_HPP
+
+#include <mlpack/core.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/methods/ann/layer/layer_types.hpp>
+#include <mlpack/methods/ann/init_rules/random_init.hpp>
+#include <mlpack/methods/ann/loss_functions/binary_cross_entropy_loss.hpp>
+
+namespace mlpack {
+namespace models {
+
+/**
+ * Definition of a VGG CNN.
```

Comment:
Actually it's probably better to put the paper reference here with the class, since users probably will look for documentation for the class and not necessarily the header file.

Do you think you can add a note on how `VGGVersion` relates to Table 1 in the paper?  i.e. `11` is equivalent to the `A` network, etc., etc.

Reply:
Idts they will look for paper here.

Reply:
It depends on the tool they are using; if they are just looking through files, I agree, but if they are using some IDE that points them to the class documentation, they will end up looking at this particular comment.  It is a minor issue, so if you feel strongly about it, it's okay to leave it how it is, but in my opinion it would be better placed here. :+1: